apis/keda/v1alpha1/scaledjob_types.go

```diff
@@ -114,19 +119,23 @@ func init() {
 // MaxReplicaCount returns MaxReplicaCount
 func (s ScaledJob) MaxReplicaCount() int64 {
 	if s.Spec.MaxReplicaCount != nil {
+		if s.Spec.MinReplicaCount != nil && *s.Spec.MinReplicaCount > *s.Spec.MaxReplicaCount {
+			return int64(*s.Spec.MaxReplicaCount)
+		}
 		return int64(*s.Spec.MaxReplicaCount) - s.MinReplicaCount()
 	}
 
-	return 100
+	return defaultScaledJobMaxReplicaCount
 }
 
 // MinReplicaCount returns MinReplicaCount
 func (s ScaledJob) MinReplicaCount() int64 {
 	if s.Spec.MinReplicaCount != nil {
-		if *s.Spec.MinReplicaCount > *s.Spec.MaxReplicaCount {
+		if s.Spec.MaxReplicaCount != nil &&
+			*s.Spec.MinReplicaCount > *s.Spec.MaxReplicaCount {
```

Comment:
May want to check if s.Spec.MinReplicaCount is nil before comparing to *s.Spec.MaxReplicaCount